src/full/Agda/TypeChecking/Rules/Term.hs

```diff
@@ -1558,12 +1558,11 @@ inferExprForWith (Arg info e) = verboseBracket "tc.with.infer" 20 "inferExprForW
     reportSDoc "tc.with.infer" 20 $ "inferExprforWith " <+> prettyTCM e
     reportSLn  "tc.with.infer" 80 $ "inferExprforWith " ++ show (deepUnscope e)
     traceCall (InferExpr e) $ do
-      -- With wants type and term fully instantiated!
-      (v, t) <- instantiateFull =<< inferExpr e
```

Comment:
Even though we `normalise` later, we could need this `instantiateFull` since it does eta-contraction, which normalisation does not do.